Game/Source/GDKTestGyms/Private/MazeGenerator.cpp

```diff
@@ -0,0 +1,171 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+
+#include "MazeGenerator.h"
+#include "Components/InstancedStaticMeshComponent.h"
+#include "SpatialGDK/Public/Utils/SpatialLatencyPayload.h"
+#include "Kismet/KismetMathLibrary.h"
+#include "Kismet/KismetArrayLibrary.h"
+
+DEFINE_LOG_CATEGORY(LogMapGenerator);
+
+// Sets default values
+AMazeGenerator::AMazeGenerator()
+{
+ 	// Set this actor to call Tick() every frame.  You can turn this off to improve performance if you don't need it.
+	PrimaryActorTick.bCanEverTick = false;
+
+	Seed = 0;
+	WidthInMetres = 150;
+	LengthInMetres = 100;
+	Rows = 2;
+	Cols = 3;
+	DefaultWallScale = FVector(1, 1, 2.5);
+	CreateWallWeight = 1;
+
+	Walls = CreateDefaultSubobject<UInstancedStaticMeshComponent>(TEXT("Walls"));
+	SetRootComponent(Walls);
+	Walls->SetMobility(EComponentMobility::Static);
+}
+
+// Called when the game starts or when spawned
+void AMazeGenerator::BeginPlay()
+{
+	Super::BeginPlay();
+	
+}
+
+void AMazeGenerator::AddWall(float X, float Y, float Length, bool bRotate)
+{
+	FTransform Transform;
+	Transform.SetLocation(FVector(X * 100.0, Y * 100.0, DefaultWallScale.Z * 50.0));
+	Transform.SetRotation(bRotate ?  FRotator(0, 90, 0).Quaternion() : FQuat::Identity);
+	Transform.SetScale3D(FVector(Length + 1, DefaultWallScale.Y, DefaultWallScale.Z));
```

Comment:
`+1` ?

Reply:
Good spot, should be DefaultWallScale.Y to make the walls extend into each other correctly.